ADPhantomApp/src/ADPhantom.cpp

```diff
@@ -2227,6 +2237,37 @@ asynStatus ADPhantom::writeInt32(asynUser *pasynUser, epicsInt32 value)
       status |= setCameraParameter("auto.bref", value);
     } 
   }
+  else if (function == PHANTOM_DataFormat_){
+    //Update the selected token and associated bit depth
+      if (value==0){
+        bitDepth_ = 8;
+        phantomToken_ = "8";
+      }
+      else if (value==1){
+        bitDepth_ = 8;
+        phantomToken_ = "8R";
+      }
+      else if (value==2){
+        bitDepth_ = 16;
+        phantomToken_ = "P16";
+      }
+      else if (value==3){
+        bitDepth_ = 16;
+        phantomToken_ = "P16R";
+      }
+      else if (value==4){
+        bitDepth_ = 10;
+        phantomToken_ = "P10";
+      }
+      else if (value==5){
+        bitDepth_ = 12;
+        phantomToken_ = "P12L";
+        printf("P12L selected!");
+      }
+      else{
```

Comment:
I'd probably then set status to asynError, that way the PV will then be reset to what it was previously (a few lines below here). This also represents what actually happens since the bitdepth and phantomToken won't change from their previous values.